myceliumd/src/cli/peer.rs

```diff
@@ -0,0 +1,97 @@
+use mycelium::peer_manager::PeerStats;
+use mycelium_api::AddPeer;
+use prettytable::{row, Table};
+use std::net::SocketAddr;
+use tracing::{debug, error};
+
+/// List the peers the current node is connected to
+pub async fn list_peers(server_addr: SocketAddr) -> Result<(), Box<dyn std::error::Error>> {
+    // Make API call
+    let request_url = format!("http://{server_addr}/api/v1/admin/peers");
+    match reqwest::get(&request_url).await {
+        Err(e) => {
+            error!("Failed to retrieve peers");
+            return Err(e.into());
+        }
+        Ok(resp) => {
+            debug!("Listing connected peers");
+            match resp.json::<Vec<PeerStats>>().await {
+                Err(e) => {
+                    error!("Failed to load response json: {e}");
+                    return Err(e.into());
+                }
+                Ok(peers) => {
+                    let mut table = Table::new();
+                    table.add_row(row![
+                        "Protocol",
+                        "Socket",
+                        "Type",
+                        "Connected",
+                        "Rx total",
+                        "Tx total"
+                    ]);
+                    for peer in peers.iter() {
+                        table.add_row(row![
+                            peer.endpoint.proto(),
+                            peer.endpoint.address(),
+                            peer.pt,
+                            peer.connection_state,
+                            peer.rx_bytes,
+                            peer.tx_bytes
```

Comment:
These raw values will quickly become a bit hard to read. Instead lets format them as `113.012 KiB`, `16.9 MiB`, etc